bot/exts/evergreen/emoji_count.py

```diff
@@ -0,0 +1,89 @@
+        self.bot = bot
+
+    def embed_builder(self, emoji: dict) -> discord.Embed:
+        """Generates an embed with the emoji names and count."""
+        embed = discord.Embed()
+        embed.color = Colours.orange
+        embed.title = "Emoji Count"
+        embed.timestamp = datetime.datetime.utcnow()
+        if len(emoji) == 1:
+            for key, value in emoji.items():
+                embed.description = f"There are **{len(value)}** emojis in the **{key}** category"
+                embed.set_thumbnail(url=random.choice(value).url)
+        else:
+            msg = ''
+            for key, value in emoji.items():
+                emoji_choice = random.choice(value)
+                emoji_info = f'There are **{len(value)}** emojis in the **{key}** category\n'
+                msg += f'<:{emoji_choice.name}:{emoji_choice.id}> {emoji_info}'
+            embed.description = msg
+        log.trace('Emoji embed built')
+        return embed
+
+    @staticmethod
+    def generate_invalid_embed(ctx: commands.Context) -> discord.Embed:
+        """Genrates error embed."""
+        embed = discord.Embed()
+        embed.color = Colours.soft_red
+        embed.title = random.choice(ERROR_REPLIES)
+        emoji_dict = {}
+        for emoji in ctx.guild.emojis:
+            emoji_dict[emoji.name.split("_")[0]] = []
+        error_comp = ', '.join(key for key in emoji_dict.keys())
+        embed.description = f"These are the valid categories\n```{error_comp}```"
+        log.trace("Error embed built")
+        return embed
+
+    def emoji_list(self, ctx: commands.Context, categories: dict) -> Dict:
+        """Generates an embed with the emoji names and count."""
+        out = {category: [] for category in categories}
+
+        for emoji in ctx.guild.emojis:
+            category = emoji.name.split('_')[0]
+            if category in out:
+                out[category].append(emoji)
+        log.trace("Emoji dict built")
+        return out
+
+    @commands.command(name="emoji_count", aliases=["ec"])
+    async def ec(self, ctx: commands.Context, *, emoji: str = None) -> Optional[str]:
+        """Returns embed with emoji category and info given by the user."""
+        emoji_dict = {}
+        for a in ctx.guild.emojis:
+            if emoji is None:
+                log.trace("Emoji Category not provided by the user")
+                emoji_dict.update({a.name.split("_")[0]: []})
+            elif a.name.split("_")[0] in emoji:
+                log.trace("Emoji Category provided by the user")
+                emoji_dict.update({a.name.split("_")[0]: []})
+        emoji_dict = self.emoji_list(ctx, emoji_dict)
+        if len(emoji_dict) == 0:
+            embed = self.generate_invalid_embed(ctx)
+            log.trace("Error embed received")
+        else:
+            embed = self.embed_builder(emoji_dict)
+            log.trace("Emoji embed received")
+        await ctx.send(embed=embed)
+        log.trace("Embed sent")
```

Comment:
This is too pointless logging.

Reply:
thats y i am using trace for logging

Reply:
Even when trace, for developers this add a lot of log messages to console, what maybe makes like error searching harder.

Reply:
ok